### PR TITLE
Add validation for branch protection

### PR DESCRIPTION
**Pull request by Donnerbart**
https://hivemq.kanbanize.com/ctrl_board/22/cards/27627/details/

### CODE REVIEW

File: release/validate-branch-protection.sh [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+      EXPECTED_CHECKS+=("$job_name ($test_plan)")
+    done <<< "${JOB_TEST_PLANS[$job_name]}"
+  done
+done
+echo
+
+# convert array to newline-separated string and sort it for comparison
+EXPECTED_CHECKS_STRING=$(printf "%s\n" "${EXPECTED_CHECKS[@]}" | sort)
+
+# get required checks for both branches
```
Comment (afalhambra-hivemq) [inline release/validate-branch-protection.sh:61]:
```suggestion
# get required checks for both the branch
```

File: release/validate-branch-protection.sh [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+REQUIRED_CHECKS=$(gh api -H "Accept: application/vnd.github.v3+json" "/repos/$OWNER/$REPO/branches/$BRANCH/protection" | jq -r '.required_status_checks.contexts | @csv' | tr ',' '\n' | tr -d '"' | sort)
```
Comment (afalhambra-hivemq) [inline release/validate-branch-protection.sh:62]: I'm not sure if the `jq` binary is also available with each Linux or Mac distribution. May be worth adding another check to the beginning of the script.

File: release/validate-branch-protection.sh [hunk 3]
```diff
@@ -0,0 +1,60 @@
+#!/bin/bash
```
Comment (afalhambra-hivemq) [inline release/validate-branch-protection.sh]:
```suggestion
#!/usr/bin/env bash
```

**Review by afalhambra-hivemq — APPROVED** [review]
Nice improvements for the release automation!

File: github-release-note-updater/src/main/java/com/hivemq/Main.java [hunk 4]
```diff
@@ -0,0 +1,223 @@
...
+        jCommander.parse(args);
+        if (arguments.help) {
+            jCommander.usage();
+            System.exit(0);
+        }
+
+        final var chartsPath = Path.of(arguments.path, "charts.json");
+        final var releasesPath = Path.of(arguments.path, "releases.json");
+        if (Files.notExists(chartsPath)) {
+            System.err.println("Charts path does not exist.");
```
[the remark this answers is not in the document]
Comment (Donnerbart) [reply github-release-note-updater/src/main/java/com/hivemq/Main.java:57]: Note: `printf()` doesn't automatically print a newline. This needs a `%n` at the end of the string.

File: github-release-note-updater/src/main/java/com/hivemq/Main.java [hunk 5]
```diff
@@ -0,0 +1,223 @@
...
+                final var exitCode = execute(executorService,
+                        arguments.gitHubCliPath,
+                        "release",
+                        "edit",
+                        releaseTag,
+                        "--repo",
+                        "hivemq/helm-charts",
+                        "--notes",
+                        releaseNote);
+                System.out.println(releaseTag + " -> " + (exitCode == 0 ? "SUCCESS" : "FAILURE"));
```
[the remark this answers is not in the document]
Comment (Donnerbart) [reply github-release-note-updater/src/main/java/com/hivemq/Main.java:136]: I'll add a `boolean success`, so we don't stop at the first failure, but report all updates and then `System.exit(1)` if there was a failure.

File: github-release-note-updater/src/main/java/com/hivemq/Main.java [hunk 6]
```diff
@@ -0,0 +1,223 @@
...
+            final var previousChart = i == 0 ? null : charts.get(i - 1);
+            final var wasChartUpdated = previousChart == null || !previousChart.appVersion().equals(chart.appVersion());
+            final var releaseTag = String.format("%s-%s", chart.name(), chart.version());
+            final var otherReleaseOptional = getMatchingRelease(releases, releaseTag, matchingReleasePattern);
+            if (wasChartUpdated || otherReleaseOptional.isEmpty()) {
+                // this release was triggered by an update of this chart
+                final var releaseNote = String.format(releaseNoteTemplate,
+                        chart.description(),
+                        chart.appVersion(),
+                        releaseUrlFunction.apply(chart.version()));
```
[the remark this answers is not in the document]
Comment (Donnerbart) [reply github-release-note-updater/src/main/java/com/hivemq/Main.java:164]: Aaaaaah, good catch. I'll let the script run again for all the releases to fix the broken URLs...

**Pull request merged**